tests/kernel/interrupt/src/interrupt_offload.c

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <irq_offload.h>
+#include <interrupt_util.h>
+
+#define STACK_SIZE	1024
+#define NUM_WORK	4
+
+static struct k_work offload_work[NUM_WORK];
+static struct k_work_q wq_queue;
+static K_THREAD_STACK_DEFINE(wq_stack, STACK_SIZE);
+static K_THREAD_STACK_DEFINE(tstack, STACK_SIZE);
+static struct k_thread tdata;
+
+static struct k_sem start_sem;
+static struct k_sem end_sem;
+
+/*
+ * This global variable control if the priority of offload job
+ * is greater than the original thread.
+ */
+static bool offload_job_prio_higher;
+
+static volatile int orig_t_keep_run;
+
+enum {
+	TEST_OFFLOAD_MULTI_JOBS,
+	TEST_OFFLOAD_IDENTICAL_JOBS
+};
+
+struct interrupt_param {
+	struct k_work *work;
+};
+
+static struct interrupt_param irq_param;
+
+/* thread entry of doing offload job */
+static void entry_offload_job(struct k_work *work)
+{
+	if (offload_job_prio_higher) {
+		/*TESTPOINT: offload thread run right after irq end.*/
+		zassert_equal(orig_t_keep_run, 0,
+			"the offload did not run immediately.");
+	} else {
+		/*TESTPOINT: original thread run right after irq end.*/
+		zassert_equal(orig_t_keep_run, 1,
+			"the offload did not run immediately.");
+	}
+
+	k_sem_give(&end_sem);
+}
+
+/* offload work to work queue */
+void isr_handler(const void *param)
+{
+	struct k_work *work = ((struct interrupt_param *)param)->work;
+
+	zassert_not_null(work, "NULL");
```

Comment:
Could you give more details for that message? Message "NULL" is not descriptive.

Reply:
Got it fixed, thanks!